Sources/StreamChatSwiftUI/ViewFactory.swift

```diff
@@ -58,6 +58,19 @@ public protocol ViewFactory: AnyObject {
         trailingSwipeLeftButtonTapped: @escaping (ChatChannel) -> Void,
         leadingSwipeButtonTapped: @escaping (ChatChannel) -> Void
     ) -> ChannelListItemType
+    
+    associatedtype ChannelAvatarViewType: View
+    /// Creates the channel avatar view shown in the channel list, search results and the channel header.
+    /// - Parameters:
+    /// - channel: the channel where the avatar is displayed.
+    /// - showOnlineIndicator: whether the online indicator is shown.
+    /// - size: the size of the avatar.
```

Comment:
As discussed above, would be nice to introduce an Options struct, so that it is more future proof